ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/backwardsync/FinalBlockConfirmation.java

```diff
@@ -44,7 +44,9 @@ static FinalBlockConfirmation genesisConfirmation(final MutableBlockchain blockc
   }
 
   static FinalBlockConfirmation ancestorConfirmation(final MutableBlockchain blockchain) {
-    return firstHeader -> blockchain.contains(firstHeader.getParentHash());
+    return firstHeader ->
+        blockchain.contains(firstHeader.getParentHash())
+            && blockchain.getChainHeadBlockNumber() + 1 >= firstHeader.getNumber();
```

Comment:
@matkt This appears to be different from the other PR: https://github.com/hyperledger/besu/pull/4972/files#diff-0361d2ff12cd43f507bf88a3cb4d781a537ca765329e0811f60a9507527cbe13R47

Reply:
looks fine though, I will assume it's an extra safety check.

Reply:
...ah just caught up with the new comments from the older PR :)
https://github.com/hyperledger/besu/pull/4972#discussion_r1097020981